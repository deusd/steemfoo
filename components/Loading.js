import React from 'react'
import { ActivityIndicator, View, Text } from 'react-native'
import Row from './layouts/Row'

const Loading = () => (
  <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
    <Row>
      <Text style={{ marginRight: 10 }}>Loading posts</Text>
      <ActivityIndicator />
    </Row>
  </View>
)

export default Loading
