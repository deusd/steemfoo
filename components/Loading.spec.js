import 'react-native'
import { shallow } from 'enzyme'
import React from 'react'
import { ActivityIndicator } from 'react-native'
import Loading from './Loading'

describe('Loading component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Loading />)
  })

  it('should render with no styles', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have an activity indicator', () => {
    expect(wrapper.contains(<ActivityIndicator />)).toBe(true)
  })
})
