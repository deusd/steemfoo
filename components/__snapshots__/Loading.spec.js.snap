// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render with no styles 1`] = `
<ForwardRef(View)
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Row>
    <ForwardRef(Text)
      style={
        Object {
          "marginRight": 10,
        }
      }
    >
      Loading posts
    </ForwardRef(Text)>
    <ForwardRef(ActivityIndicator)
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </Row>
</ForwardRef(View)>
`;
